rbac/common/util/__init__.py

```diff
@@ -31,3 +31,37 @@ def get_environment(name, default):
     if value is None or not value:
         return default
     return value
+
+
+def bytes_from_hex(value):
+    """A version of bytes.fromhex that will return None if
+    passed None instead of an expected str exception"""
+    if not value:
+        return None
+    return bytes.fromhex(value)
+
+
+def has_duplicates(anylist):
+    """Returns whether anylist has duplicates"""
+    if not isinstance(anylist, list):
+        raise ValueError(
+            "has_duplicates expected a list, got {}\n{}".format(type(anylist), anylist)
+        )
+    return bool(len(anylist) != len(set(anylist)))
+
+
+def duplicates(anylist):
+    """Returns the duplicate values in any list"""
+    if not isinstance(anylist, list):
+        raise ValueError(
+            "duplicates expected a list, got {}\n{}".format(type(anylist), anylist)
+        )
+    # return list(set([value for value in anylist if anylist.count(value) > 1])) # inefficient
```

Comment:
Yes, it's the obvious way to accomplish the task, and what you would normally use. It's not used here, because as a generic utility function you want to support future use cases that may included larger datasets.

Reply:
It's not used. The part about it being obvious and what "you would normally use" is certainly possible, but an assumption. If it is that important: Please add more context to the comment (why it is inefficient relative to the decision you went with, how it doesn't scale, etc)